app/grandchallenge/reader_studies/models.py

```diff
@@ -251,6 +251,28 @@ def hanging_list_images(self):
 
         return hanging_list_images
 
+    @property
+    def image_groups(self):
+        return [sorted(x.values()) for x in self.hanging_list]
+
+    def add_ground_truth(self, *, data, user):
+        for gt in data:
+            images = self.images.filter(
+                name__in=gt["images"].split(";")
+            ).values_list("id", flat=True)
```

Comment:
Have we checked that these images belong to this study?

Reply:
Or is that done in L267 on answer create?

Reply:
That is done in the field clean method on the form (https://github.com/comic/grand-challenge.org/pull/1094/files#diff-8f1ba48a837c1a7befad0375e2fc0740R235). This checks if the images provided in the csv match with the hanging list.